### PR TITLE
Refined backend of StudentCourseDetailsPageAction Slice

### PR DESCRIPTION
**Pull request by bamboo3250**
refined StudentCourseDetailsPageAction.java
refined StudentCourseDetailsPageActionTest.java
refined StudentCourseDetailsPageUiTest.java
refined StudentCourseDetailsPageAction.java

Added getTeamForStudent function to StudentsLogic
Added a test case for the function

### CODE REVIEW

File: src/main/java/teammates/logic/api/Logic.java [hunk 1]
```diff
@@ -1075,6 +1075,15 @@ public SectionDetailsBundle getSectionForCourse(String section, String courseId)
         Assumption.assertNotNull(ERROR_NULL_PARAMETER, courseId);
         return coursesLogic.getTeamsForCourse(courseId);
     }
+    
+    /**
+     * Preconditions: <br>
+     * * All parameters are non-null.
+     */
+    public TeamDetailsBundle getTeamForStudent(StudentAttributes student) throws EntityDoesNotExistException {
```
Comment (damithc) [inline src/main/java/teammates/logic/api/Logic.java]: `getTeamDeatailsForStudent`?

Comment (bamboo3250) [inline src/main/java/teammates/logic/api/Logic.java]: That sounds more precise. I will rename it.

**Comment by wkurniawan07** [comment]
Dude you forgot to change the method name `getTeamDetailsForStudent` in the test file. Now eclipse won't even run the project.

**Comment by wkurniawan07** [comment]
Anyway, I'm going to merge mine soon so I'll just fix it myself.

**Comment by bamboo3250** [comment]
Oops, I forgot to commit that part after running the test suit.
Anyway, thanks. Too much caffeine for today =.=

**Pull request merged**